common/src/main/java/com/wynntils/models/character/CharacterModel.java

```diff
@@ -213,13 +215,20 @@ private void scanCharacterInfo() {
         }
 
         // Scan guild container, if the player is in a guild
-        queryBuilder.conditionalThen(
-                // The guild name has already been parsed at this point
-                (container) -> !Models.Guild.getGuildName().isEmpty(),
-                // Open Guild Menu
-                QueryStep.clickOnSlot(GUILD_MENU_SLOT)
-                        .expectContainerTitle(ContainerModel.GUILD_MENU_NAME)
-                        .processIncomingContainer(Models.Guild::parseGuildContainer));
+        // Upon execution the guild name has already been parsed
+        ContainerPredicate guildMember =
+                (container) -> !Models.Guild.getGuildName().isEmpty();
+        queryBuilder
+                .conditionalThen(
+                        guildMember,
+                        QueryStep.clickOnSlot(GUILD_MENU_SLOT)
+                                .expectContainerTitle(ContainerModel.GUILD_MENU_NAME)
+                                .processIncomingContainer(Models.Guild::parseGuildContainer))
+                .conditionalThen(
+                        guildMember,
+                        QueryStep.clickOnSlot(GuildModel.DIPLOMACY_MENU_SLOT)
+                                .expectContainerTitle(ContainerModel.GUILD_DIPLOMACY_MENU_NAME)
+                                .processIncomingContainer(Models.Guild::parseDiplomacyContainer));
```

Comment:
As I've mentioned before, these queries are starting to get a bit too much. In fact, playing with the last patch already felt slower than before. However, I do understand that for the guild functionality, we need this parsing to happen. Give me a bit to think about how we can make this optional, or do this less times, etc.

Reply:
There is definitely a way to skip this container now that I think of it.
If a guild has no allies (which can be parsed form the previous container) this scan shouldnt be necessary. However due to Wynncraft itself not showing sent tributes in the previous menu (in contrast to recieved ones) we cant rule out tributes affecting global usage.

Reply:
If it werent for https://forums.wynncraft.com/threads/guilds.281813/page-3#post-3658780
we could also skip the parsing of that container if the tooltip says "No tributes sent or received".

Reply:
Nothing you can do about that. But I can offer you something to work on, that can "compensate" for this additional tracking. Everything that is parsed from `parseCratesBombsCosmeticsContainer` can't really change during gameplay, except the silverbull rank. So that parsing only needs to happen once per launch. The problem is detecting when a player purchased silverbull. Neither you or me has the regexes for the chat messages... So I can't request you to do that change here, however, if in some way, you can do it, that would be the best.

Reply:
Otherwise, I'll think about creating configs, or something else to compensate for the speed..

Reply:
> Nothing you can do about that. But I can offer you something to work on, that can "compensate" for this additional tracking. Everything that is parsed from `parseCratesBombsCosmeticsContainer` can't really change during gameplay, except the silverbull rank. So that parsing only needs to happen once per launch. The problem is detecting when a player purchased silverbull. Neither you or me has the regexes for the chat messages... So I can't request you to do that change here, however, if in some way, you can do it, that would be the best.

The problem would still be that while you are disconnected or in class selection/hub you wouldn't be able to parse chat or other changes.

How about having a threshold of time that skips guild and silverbull parsing if the time delta between logging out and back in is short enough (i. e. short enough for skipping it on entering wars and /switch).

Reply:
> The problem would still be that while you are disconnected or in class selection/hub you wouldn't be able to parse chat or other changes.

You don't get these messages when disconnected, they happen on join. Store messages work like this, never seen silverbull though.

Reply:
Ooooh, I in fact might have a regex for that then

Reply:
Okay: good and bad news: There is a chat message, but I do not know if it behaves similarily to the purchase message, as it is in different nature to regular shop messages:

```
[21:43:21] [Render thread/INFO]: [CHAT] À
[21:43:21] [Render thread/INFO]: [CHAT]          &3Welcome to the &b✮ Silverbull Trading Company&3!
[21:43:21] [Render thread/INFO]: [CHAT]            &7Your subscription has been activated.
[21:43:21] [Render thread/INFO]: [CHAT] À
[21:43:21] [Render thread/INFO]: [CHAT]                         &a✔ &fLower market taxes
[21:43:21] [Render thread/INFO]: [CHAT]                         &a✔ &fGuild boost ability
[21:43:21] [Render thread/INFO]: [CHAT]                         &a✔ &fDouble daily rewards
[21:43:21] [Render thread/INFO]: [CHAT]                         &a✔ &fFree monthly shares
[21:43:21] [Render thread/INFO]: [CHAT]                       &f&oAnd more unlocked!
[21:43:21] [Render thread/INFO]: [CHAT] À
```

Reply:
I think it's fine to assume, it is.

Reply:
I have added the requested changes, however I'm not able to test them myself

Reply:
Can you, @ShadowCat117?

Reply:
I have never had a Silverbull subscription. Let me know in Discord if you need my help.

Reply:
> I have never had a Silverbull subscription. Let me know in Discord if you need my help.

Me neither. If we really want to test this, we can get Scyu to gift either of us a month of silverbull subscription, from the store.

Reply:
Probably better you get it, I guess you could actually put it to some good use. :)